scylla-cql/src/frame/request/auth_response.rs

```diff
@@ -1,37 +1,25 @@
 use crate::frame::frame_errors::ParseError;
 use bytes::BufMut;
-use std::convert::TryInto;
 
 use crate::frame::request::{Request, RequestOpcode};
-use crate::frame::Authenticator;
 
 // Implements Authenticate Response
 pub struct AuthResponse {
-    pub username: Option<String>,
-    pub password: Option<String>,
-    pub authenticator: Authenticator,
+    pub response: Option<Vec<u8>>,
 }
 
 impl Request for AuthResponse {
     const OPCODE: RequestOpcode = RequestOpcode::AuthResponse;
 
     fn serialize(&self, buf: &mut impl BufMut) -> Result<(), ParseError> {
-        if self.username.is_none() || self.password.is_none() {
-            return Err(ParseError::BadDataToSerialize(
-                "Bad credentials: username or password missing. You can use SessionBuilder::user(\"user\", \"pass\") to provide credentials.".to_string(),
-            ));
-        }
-        let username_as_bytes = self.username.as_ref().unwrap().as_bytes();
-        let password_as_bytes = self.password.as_ref().unwrap().as_bytes();
+        if let Some(bytes) = self.response.clone() {
+            buf.put_i32(bytes.len().try_into()?);
```

Comment:
`self.response` is, in the terminology of the CQL protocol spec, a `[bytes]` object. `[bytes]` is serialized as follows:

- if null, then serialize `-1` as a little-endian `i32` and that's it,
- if not null, then serialize its length as a little-endian `i32` and then just serialize the contents.

We have a module `scylla-cql/src/frame/types.rs` which provides functions for serialization and deserialization of CQL protocol spec objects. You need the `types::write_bytes_opt` function - which does not exist, but I'm sure you can add it yourself after looking at how other functions are implemented in this file.